web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -538,21 +540,24 @@
   $: if (!$user) {
     shouldShowShareModal = false;
   }
-</script>
 
-<svelte:window
-  use:shortcuts={[
+  let shortcutOptions: ShortcutOptions<unknown>[] = [
     { shortcut: { key: 'a', shift: true }, onShortcut: toggleAssetArchive },
     { shortcut: { key: 'ArrowLeft' }, onShortcut: () => navigateAsset('previous') },
     { shortcut: { key: 'ArrowRight' }, onShortcut: () => navigateAsset('next') },
     { shortcut: { key: 'd', shift: true }, onShortcut: () => downloadFile(asset) },
-    { shortcut: { key: 'Delete' }, onShortcut: () => trashOrDelete(false) },
     { shortcut: { key: 'Delete', shift: true }, onShortcut: () => trashOrDelete(true) },
     { shortcut: { key: 'Escape' }, onShortcut: closeViewer },
     { shortcut: { key: 'f' }, onShortcut: toggleFavorite },
     { shortcut: { key: 'i' }, onShortcut: toggleDetailPanel },
-  ]}
-/>
+  ];
+
+  if (!asset.isTrashed) {
+    shortcutOptions.push({ shortcut: { key: 'Delete' }, onShortcut: () => trashOrDelete(false) });
```

Comment:
This one is not tested. I could - but I would refactor the callbacks to live outside the component, for easier testing. Shall I?